src/mapper/cmd/main.go

```diff
@@ -99,25 +100,28 @@ func main() {
 	telemetrysender.SetGlobalContextId(telemetrysender.Anonymize(kubeSystemUID))
 
 	// start API server
-	e := echo.New()
-	e.GET("/healthz", func(c echo.Context) error {
+	mapperServer := echo.New()
+	mapperServer.GET("/healthz", func(c echo.Context) error {
 		return c.NoContent(http.StatusOK)
 	})
-	e.Use(middleware.Logger())
-	e.Use(middleware.CORS())
-	e.Use(middleware.RemoveTrailingSlash())
+	mapperServer.Use(middleware.Logger())
+	mapperServer.Use(middleware.CORS())
+	mapperServer.Use(middleware.RemoveTrailingSlash())
 	initCtx, cancelFn := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancelFn()
 	mgr.GetCache().WaitForCacheSync(initCtx) // needed to let the manager initialize before used in intentsHolder
 
 	intentsHolder := intentsstore.NewIntentsHolder()
 
 	resolver := resolvers.NewResolver(kubeFinder, serviceidresolver.NewResolver(mgr.GetClient()), intentsHolder)
-	resolver.Register(e)
+	resolver.Register(mapperServer)
 
-	signalHandlerCtx, signalHandlerCancel := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
-	defer signalHandlerCancel()
-	cloudClient, cloudEnabled, err := cloudclient.NewClient(signalHandlerCtx)
+	metricsServer := echo.New()
+
+	metricsServer.GET("/metrics", echoprometheus.NewHandler())
+
+	errgrp, errGroupCtx := errgroup.WithContext(signals.SetupSignalHandler())
```

Comment:
This will cause immediate panic, since you can call `signals.SetupSignalHandler` only once

Reply:
whoops